python/lsst/ctrl/oods/validator.py

```diff
@@ -0,0 +1,149 @@
+    """
+
+    def __init__(self, config, verbose=False):
+        """initialize
+        @param config: a configuration data structure
+        @param verbose: whether to emit messages when errors are found.
+        """
+        self.oodsConfig = config
+        self.isValid = False
+        self.verbose = verbose
+        self.missingElements = []
+        self.missingValues = []
+
+    def verify(self):
+        """Validate a configuration, emitting messages about
+        """
+        self.isValid = True
+        self.missingElements = []
+        self.missingValues = []
+
+        if self.oodsConfig is None:
+            self.isValid = False
+            self.missingElement("ingester")
+            self.missingElement("cacheCleaner")
+            return
+
+        configName = "ingester"
+        if configName not in self.oodsConfig:
+            self.missingElement("ingester")
+
+        else:
+            ingesterConfig = self.oodsConfig[configName]
+            if "directories" in ingesterConfig:
+                dirs = ingesterConfig["directories"]
+                if dirs is None:
+                    self.missingValue("ingester:directories")
+                elif len(dirs) == 0:
+                    self.missingValue("ingester:directories")
+            else:
+                self.missingElement("ingester:directories")
+            if "butler" in ingesterConfig:
+                butlerConfig = ingesterConfig["butler"]
+                if "class" in butlerConfig:
+                    classConfig = butlerConfig["class"]
+                    if "import" not in classConfig:
+                        self.missingElement("butler:class:import")
+                    if "name" not in classConfig:
+                        self.missingElement("butler:class:name")
+                else:
+                    self.missingElement("butler:class")
+                if "repoDirectory" not in butlerConfig:
+                    self.missingElement("butler:repoDirectory")
+            else:
+                self.missingElement("ingester:butler")
+
+            if "batchSize" not in ingesterConfig:
+                self.missingElement("ingester:batchSize")
+
+            self.checkIntervalBlock("scanInterval", configName, ingesterConfig)
+
+        configName = "cacheCleaner"
+        if configName not in self.oodsConfig:
+            self.missingElement(configName)
+        else:
+            cacheConfig = self.oodsConfig[configName]
+            name = "directories"
+            if name in cacheConfig:
+                dirs = cacheConfig[name]
+                if dirs is None:
+                    self.missingValue("%s:%s" % (configName, name))
+            else:
+                self.missingElement("%s:directories" % configName)
+
+            self.checkIntervalBlock("scanInterval", configName, cacheConfig)
+            self.checkIntervalBlock("filesOlderThan", configName, cacheConfig)
+            name = "directoriesEmptyForMoreThan"
```

Comment:
Um, why is this different than the previous two?

Reply:
80 character line limit

Reply:
Um,

```
self.checkIntervalBlock("directoriesEmptyForMoreThan",
    configName, cacheConfig")
```
should be fine.